pallets/utxo/src/tests.rs

```diff
@@ -601,3 +543,667 @@ fn test_send_to_address() {
+            outputs: vec![TransactionOutput::new_with_data(
+                0,
+                H256::from(alice_pub_key),
+                OutputData::TokenTransferV1 {
+                    token_id: token_id.clone(),
+                    amount: 1_000_000_001,
+                },
+            )],
+            time_lock: Default::default(),
+        }
+        .sign_unchecked(&[token_utxo.clone()], 0, &karl_pub_key);
+        frame_support::assert_err_ignore_postinfo!(
+            Utxo::spend(Origin::signed(H256::zero()), tx),
+            "output value must not exceed input value"
+        );
+
+        // Let's send a big amount of MLT with the correct tokens
+        let tx = Transaction {
+            inputs: vec![TransactionInput::new_empty(token_utxo_hash)],
+            outputs: vec![TransactionOutput::new_with_data(
+                1_000_000_000,
+                H256::from(alice_pub_key),
+                OutputData::TokenTransferV1 {
+                    token_id: token_id.clone(),
+                    amount: 1_000_000_000,
+                },
+            )],
+            time_lock: Default::default(),
+        }
+        .sign_unchecked(&[token_utxo.clone()], 0, &karl_pub_key);
+        frame_support::assert_err_ignore_postinfo!(
+            Utxo::spend(Origin::signed(H256::zero()), tx),
+            "output value must not exceed input value"
+        );
+
+        // should be success
+        let tx = Transaction {
+            inputs: vec![TransactionInput::new_empty(token_utxo_hash)],
+            outputs: vec![TransactionOutput::new_with_data(
+                0,
+                H256::from(alice_pub_key),
+                OutputData::TokenTransferV1 {
+                    token_id: token_id.clone(),
+                    amount: 1_000_000_000,
+                },
+            )],
+            time_lock: Default::default(),
+        }
+        .sign_unchecked(&[token_utxo], 0, &karl_pub_key);
+        assert_ok!(Utxo::spend(Origin::signed(H256::zero()), tx));
+    });
+}
+
+#[test]
+fn checking_nft_transferring() {
+    let (mut test_ext, alice_pub_key, karl_pub_key) = new_test_ext_and_keys();
+    test_ext.execute_with(|| {
+        let token_id = TokenId::new_asset(H256::random());
+        // Alice issue 1000 MLS-01, and send them to Karl and the rest back to herself
+        let (utxo0, input0) = tx_input_gen_no_signature();
+        let tx = Transaction {
+            inputs: vec![input0],
+            outputs: vec![
+                TransactionOutput::new_pubkey(90, H256::from(alice_pub_key)),
+                TransactionOutput::new_with_data(
+                    10,
+                    H256::from(karl_pub_key),
+                    OutputData::NftMintV1 {
+                        token_id: token_id.clone(),
+                        data_hash: NftDataHash::Hash32([7; 32]),
+                        metadata_uri: "facebook.com".as_bytes().to_vec(),
```

Comment:
Let's change that to mintlayer.org because fuck facebook

Reply:
Haha, amazing!

Reply:
Done